src/pybind/mgr/devicehealth/module.py

```diff
@@ -489,37 +525,39 @@ def put_device_metrics(self, ioctx: rados.Ioctx, devid: str, data: Any) -> None:
                 self.log.debug(f"removing {devid} wear level")
                 self.set_device_wear_level(devid, -1.0)
 
+    def _t2epoch(self, t: Optional[str]) -> int:
+        if t is None:
+            return 0
+        else:
+            return int(datetime.strptime(t, TIME_FORMAT).strftime("%s"))
+
     def _get_device_metrics(self, devid: str,
                             sample: Optional[str] = None,
                             min_sample: Optional[str] = None) -> Dict[str, Dict[str, Any]]:
         res = {}
-        ioctx = self.open_connection(create_if_missing=False)
-        if not ioctx:
-            return {}
-        with ioctx:
-            with rados.ReadOpCtx() as op:
-                omap_iter, ret = ioctx.get_omap_vals(op, min_sample or '', sample or '',
-                                                     MAX_SAMPLES)  # fixme
-                assert ret == 0
+
+        SQL = """
+        SELECT time, raw_smart
+            FROM DeviceHealthMetrics
+            WHERE devid = ? AND (time = ? OR ? <= time)
```

Comment:
I think that if `min_sample` is set, then we're looking for records that are newer than it;
therefore:
`? >= time`

Reply:
That would select all records older than `time`. I don't think so?

Reply:
sorry, I read it the other way around, you're right!